statemachine/errors_test.go

```diff
@@ -42,3 +45,24 @@ func TestDefaultErrHandler(t *testing.T) {
 		}
 	}
 }
+
+type errType1 struct{ error }
+type errType2 struct{ error }
+
+func TestErr(t *testing.T) {
+	err := errType1{errors.New("some underlying error")}
+	se := NewErr(err, time.Now())
+
+	// confirm se implements the error interface
+	assert.Implements(t, (*error)(nil), se)
```

Comment:
If you make this a require I believe you can ditch the panic test below:

```suggestion
	require.Implements(t, (*error)(nil), se)
```